assets/app/scripts/directives/overviewDeployment.js

```diff
@@ -27,27 +28,39 @@ angular.module('openshiftConsole')
           $scope.desiredReplicas = null;
         });
 
-        // Debounce scaling so multiple clicks within 500 milliseconds only
-        // result in one request.
+        $scope.canScale = function() {
+          if (!$scope.deploymentConfigId || $scope.deploymentConfigMissing) {
+            // Scale as replication controller.
+            return true;
+          }
+
+          if (!$scope.deploymentConfig) {
+            // Wait for the deployment config.
+            return false;
+          }
+
+          // Only allow scaling of the latest deployment.
+          return deploymentIsLatestFilter($scope.rc, $scope.deploymentConfig);
```

Comment:
We've done this all along to keep the UI responsive waiting for the debounce and rc update... We show the requested value immediately as soon as the user clicks the arrow. The only time another number would show up is if rc.spec.replicas changed to a value other than the previous or target before scaling completed.